### PR TITLE
chore: fix layer hierarchy

### PR DESCRIPTION
**Pull request by supproduction**
Moves base/form package to the platform domain.
Moves components plugin/useComponent/LazyLoad resolver from core into utilities.
Moves Type interface from core into utilities.
Fixes unit tests missing prototypes errors.

closes: [HRZ-2132](https://spryker.atlassian.net/browse/HRZ-2132)

[HRZ-2132]: https://spryker.atlassian.net/browse/HRZ-2132?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -19,27 +19,20 @@
                   "layer:platform",
                   "layer:base",
                   "layer:domain",
-                  "layer:template",
-                  "lib:experience",
-                  "lib:i18n",
-                  "lib:ui",
-                  "lib:form"
+                  "layer:template"
```
Comment (tolerants) [inline .eslintrc.json]: But it's a violation, `domain` shouldn't depend on `template` layer. So basically everything is allowed for this layer.

File: .eslintrc.json [hunk 2]
```diff
@@ -19,27 +19,20 @@
...
               {
                 "sourceTag": "layer:platform",
                 "onlyDependOnLibsWithTags": [
                   "layer:platform",
                   "layer:base",
-                  "lib:experience",
-                  "lib:i18n",
-                  "lib:ui",
-                  "lib:form"
+                  "layer:domain"
```
Comment (tolerants) [inline .eslintrc.json]: The same.

File: libs/base/di/src/models/type.ts [hunk 3]
```diff
@@ -1,6 +1,5 @@
+// TODO: remove when fix circular dependency utilities <=> di
```
Comment (tolerants) [inline libs/base/di/src/models/type.ts:1]: Let's create a ticket maybe to track it?

**Review by tolerants — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (supproduction) [reply .eslintrc.json]: we use import from core into domains `feature` definitions. So we can't omit platform, and template import we use for picking import. We can create picking exception but we can't omit platform until we move out orchestrator?

File: .eslintrc.json (hunk 2 above)
Comment (supproduction) [reply .eslintrc.json]: But all good here 
```
{
                "sourceTag": "layer:platform",
                "onlyDependOnLibsWithTags": [
                  "layer:platform",
                  "layer:base",
                  "layer:domain"
                ]
              },
```

File: .eslintrc.json [hunk 4]
```diff
@@ -51,37 +39,12 @@
               },
               // Exception rules
               {
-                "sourceTag": "lib:experience",
+                "sourceTag": "lib:picking",
```
Comment (tolerants) [inline .eslintrc.json:42]: As I understand the only issue is in BAPI feature from the `application` package. But I'm thinking if we can move them to the `auth` package.

**Review by tolerants — CHANGES_REQUESTED** [review]

**Review by tolerants — APPROVED** [review]

**Pull request merged**